pydantic/_internal/_generate_schema.py

```diff
@@ -117,6 +124,26 @@ def generate_schema(self, obj: Any) -> core_schema.CoreSchema:  # noqa: C901
             if issubclass(obj, dict):
                 return self._dict_subclass_schema(obj)
             # probably need to take care of other subclasses here
+        elif isinstance(obj, typing.TypeVar):
+            # TODO: Add a check for recursion depth exception and an explanatory re-raise
```

Comment:
(note to self -- need to delete this comment; the exception handling has already been added below)